<?xml version="1.0"?>
<launch>
    <arg name="map_path" value="$(find quad)/data/sample.pcd" />

    <group ns="drone0">
        <arg name="id" value="drone0"/>
        <node pkg="quad" type="quad_node" name="quad_node" output="screen">
            <rosparam command="load" file="$(find quad)/launch/simulation_parameters.yaml" />
            <rosparam command="load" file="$(find quad)/launch/map_parameters.yaml" />
            <param name="agent/mesh_resource" value="file://$(find quad)/meshes/fake_drone.dae" />
            <param name="agent/id" value="$(arg id)"/>
            <param name="agent/start_x" value="0.0"/>
            <param name="agent/start_y" value="0.0"/>
            <param name="agent/start_z" value="2.0"/>
            <param name="agent/yaw_offset" value="0.0"/>
            <param name="map/path" value="$(arg map_path)"/>
        </node>
    </group>

    <node pkg="quad" type="map_publisher" name="map_publisher" output="screen">
        <param name="map/path" value="$(arg map_path)"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" 
        args="-d $(find quad)/rviz/display.rviz"/>

</launch>